### ISSUE TITLE
Seeing a long hang when switching Scene Collection (Release 26.1.0)

### ISSUE THREAD

**Issue by Jazzynupe** [obsproject/obs-studio#3886]
Whenever loading the OBS Studio interface and changing the scene collection, the process just hangs and does not function until I kill the process. I have deleted everything that was in the "Plugins" folders ( "C:\Program Files\obs-studio\obs-plugins\64bit ") and ran a freshly downloaded installer again.

## Platform
Operating system and version: Windows 10 ( 10.0.21277 )
OBS Studio version: 26.1.0 (64-bit)

## Expected Behavior
When choosing a scene collection, it should switch and load these new scenes to the interface and open devices needed

## Current Behavior
Currently the interface will load fine initially, however as soon as I select a new scene collection, it hangs and freezes until I kill the process. Crash reports are not being generated when killed. If desired, I can set up a crash dump capture using Procdump.

Log file that was generated:
[2020-12-14 19-00-56.txt](https://github.com/obsproject/obs-studio/files/5693199/2020-12-14.19-00-56.txt)

## Steps to Reproduce

1. Start OBS Studio (version 26.1.0)
      a. Everything should load normally at this point
2. Once started, have a profile with multiple scene collections
      a. You should be able to choose scenes fine in the existing collection
![image](https://user-images.githubusercontent.com/16325683/102168230-c2b0d880-3e44-11eb-9ffe-f4895faaef64.png)

3. on the scene collection menu option select a different scene collection
      a. Your scene collection options should appear, however upon selecting a different one you should now be hung.
![image](https://user-images.githubusercontent.com/16325683/102168109-88dfd200-3e44-11eb-8b00-4e9d57e93833.png)

4.

## Additional information

A crash or hang was also reported by another user on Twitter as linked here (maybe unrelated since I can open and switch scenes):
https://twitter.com/DJBluePDX/status/1338620888233590784?s=20

**Comment by WizardCM** [comment]
I can't reproduce this, so it's likely one of your sources that's causing the hang (or a secondary program). I added an instance of every built-in source type. A dump would likely help, but it may be worth asking in the Discord server so you don't have to post it publicly. https://obsproject.com/discord

**Comment by Jazzynupe** [comment]
I can attach all the scenes before the upgrade. Things were switching fine before the upgrade so maybe you can see something different here. I can prepare a crash dump by tomorrow as well. They are browser scenes, 1 camera (Life Cam), and a set of program/window captures. The Intro has 1 video source as well. Not too complicated I don't think
[Red_Desktop.txt](https://github.com/obsproject/obs-studio/files/5693280/Red_Desktop.txt)
[Cool_Desktop.txt](https://github.com/obsproject/obs-studio/files/5693281/Cool_Desktop.txt)
[Plain_Desktop.txt](https://github.com/obsproject/obs-studio/files/5693282/Plain_Desktop.txt)

**Comment by Jazzynupe** [comment]
After setting up the memory dump, I ran the application again and captured a preliminary memory dump using ProcDump64 and tried to replicate.
This time the hang happened when I went to shutdown the application. Likely a source issue as you said, but I was able to duplicate it several times for a few hours before. Oh well, I will provide Thread 0 call stack for reference (64bit) but until it is more consistent, I will let this close for now

![image](https://user-images.githubusercontent.com/16325683/102173629-44f2ca00-3e50-11eb-950f-af51230b0e80.png)
